Jenkinsfile

```diff
@@ -163,58 +63,13 @@ node('vetsgov-general-purpose') {
     }
   }
 
-  stage('Build: Redirects') {
-    if (shouldBail()) { return }
-
-    try {
-      def builds = [:]
-
-      for (int i=0; i<VETSGOV_BUILDTYPES.size(); i++) {
-        def envName = VETSGOV_BUILDTYPES.get(i)
-        builds[envName] = {
-          dockerImage.inside(args) {
-            sh "cd /application && npm --no-color run build:redirects -- --buildtype=${envName}"
-            sh "cd /application && echo \"${buildDetails('buildtype': envName, 'ref': ref)}\" > build/${envName}/BUILD.txt"
-          }
-        }
-      }
-
-      parallel builds
-    } catch (error) {
-      notify()
-      throw error
-    }
-  }
-
   // Perform a build for each build type
-
-  stage('Build') {
-    if (shouldBail()) { return }
-
-    try {
-      def builds = [:]
-      def assetSource = (cmsEnv != 'none' && cmsEnv != 'live') ? ref : 'local'
-
-      for (int i=0; i<VAGOV_BUILDTYPES.size(); i++) {
-        def envName = VAGOV_BUILDTYPES.get(i)
-        builds[envName] = {
-          dockerImage.inside(args) {
-            sh "cd /application && npm --no-color run build -- --buildtype=${envName} --asset-source=${assetSource}"
-            sh "cd /application && echo \"${buildDetails('buildtype': envName, 'ref': ref)}\" > build/${envName}/BUILD.txt"
-          }
-        }
-      }
-
-      parallel builds
-    } catch (error) {
-      notify()
-      throw error
-    }
-  }
+  withCms = (cmsEnv != 'none' && cmsEnv != 'live')
+  buildUtil.build(ref, dockerImage, dockerArgs, withCms)
 
   // Run E2E and accessibility tests
   stage('Integration') {
-    if (shouldBail() || !VAGOV_BUILDTYPES.contains('vagovprod')) { return }
+    if (buildUtil.shouldBail() || !VAGOV_BUILDTYPES.contains('vagovprod')) { return }
```

Comment:
I think we could make these `VAGOV_BUILDTYPES.contains(` conditions clearer, because I had to trace back where `VAGOV_BUILDTYPES` is modified to understand what was happening there. It might be clearer to create a new lowercase variable `targetBuildtypes` to help show that this value is no longer constant. I left another comment related to this.